pkg/execution/state/redis_state/redis_state.go

```diff
@@ -1238,11 +1230,6 @@ func (m unshardedMgr) LoadEvaluablesSince(ctx context.Context, workspaceID uuid.
 		}
 	}
 
-	// GC pauses on fetch.
-	for _, pause := range expired {
```

Comment:
@tonyhb as far as I can see, we will delete expired pauses either

- when the timeout queue item is run
- when another event comes in after the grace period introduced above

I believe deleting pauses in the iterator for aggregate matching may introduce races where the timeout job hasn't run but the pause already expired.

We could also keep this if we introduced the same grace period.

Reply:
We _should_ probably delete pauses from the index to prevent fetching expired pauses over and over. We should not drop the entire pause, though.